instrumentation/sinatra/lib/opentelemetry/instrumentation/sinatra/middlewares/tracer_middleware.rb

```diff
@@ -16,17 +16,25 @@ def initialize(app)
           end
 
           def call(env)
-            @app.call(env)
+            response = @app.call(env)
           ensure
-            trace_response(env)
+            trace_response(env, response)
           end
 
-          def trace_response(env)
+          def trace_response(env, response)
             span = OpenTelemetry::Instrumentation::Rack.current_span
             return unless span.recording?
 
             span.set_attribute('http.route', env['sinatra.route'].split.last) if env['sinatra.route']
             span.name = env['sinatra.route'] if env['sinatra.route']
+
+            return if response.nil?
+
+            sinatra_response = ::Sinatra::Response.new([], response.first)
+            return unless sinatra_response.server_error?
+
+            span.record_exception(env['sinatra.error']) if env['sinatra.error']
+            span.status = OpenTelemetry::Trace::Status.error
```

Comment:
I'm not sure if we should tag the span as erroneous here since that's also done in the Rack middleware. However I've left it as-is(/was) to keep exception+status setting together

Reply:
We should _only_ tag it as an error if the response is in 5xx range

Reply:
It will only set it for a `>500`. That is done by checking `sinatra_response.server_error?`

Reply:
> It will only set it for a >500. That is done by checking sinatra_response.server_error?

Fair. 

> should we keep it here? (to keep exception and status setting together) or should we only set exception here, and set status in rack middleware?

I do not object to setting it here since it is more explicit.

I am not a sinatra user so I admit that I have not done the appropriate due diligence reviews here. 

May I ask you to share some screen shots of Jaeger with the traces before and after so we know what users should see?

Reply:
Nice, I believe setting error status and adding the exception go hand-in-hand and I like it to be explicit.

Current behavior in 0.21.4 (sets status correctly (in Rack middleware), but does not add exception):
![216688451-975a6ff5-a5b8-44a8-b770-1f34409e67ff](https://user-images.githubusercontent.com/27202540/218275909-6ffc1921-ef56-4cb9-8272-1ec96e4d995b.png)

Behavior after this PR (sets status correctly, and adds exception):
![Screenshot 2023-02-11 at 19-55-31 Jaeger UI](https://user-images.githubusercontent.com/27202540/218276120-d248a8dd-90f9-4d44-8c41-deb10c239149.png)

Essentially, this is the same behavior as in 0.21.3, but additionally it should support the use-case of @rdoherty-fastly as well